src/core/etl/src/Flow/ETL/PHP/Type/TypeFactory.php

```diff
@@ -30,7 +27,7 @@ public function getType(mixed $value) : Type
         }
 
         if (\is_object($value)) {
```

Comment:
hmm maybe `TypeDetector` would be a better name for this class, since it's not really creating anything new, it's detecting the type based on the provided value.

Reply:
Renamed.